go/enclave/components/batch_executor.go

```diff
@@ -110,6 +110,7 @@ func (executor *batchExecutor) ComputeBatch(context *BatchExecutionContext) (*Co
 
 	// we need to copy the batch to reset the internal hash cache
 	copyBatch := *batch
+	copyBatch.ResetHash()
 	copyBatch.Header.Root = stateDB.IntermediateRoot(false)
 	copyBatch.Transactions = successfulTxs
```

Comment:
Do we need to worry about `Header.TxHash` being out of sync with `copyBatch.Transactions` too?

Reply:
I don't think so. But who knows